test/jdk/com/sun/net/httpserver/HttpsParametersClientAuthTest.java

```diff
@@ -0,0 +1,513 @@
+            try {
+                final HttpClient.Builder builder = createClientBuilder();
+                // if the client is expected to present client certificates, then
+                // we construct a SSLContext which has both key and trust material.
+                // otherwise we construct a SSLContext that only has trust material
+                // and thus won't present certificates during TLS handshake
+                final SSLCtx clientSSLCtx = presentClientCerts
+                        ? keyAndTrustStoreContext() : onlyTrustStoreContext();
+                // construct the client using the SSLContext
+                try (final HttpClient client = builder.sslContext(clientSSLCtx.sslContext)
+                        .build()) {
+                    // issue a request
+                    final URI reqURI = URIBuilder.newBuilder()
+                            .scheme("https")
+                            .host(server.getAddress().getAddress())
+                            .port(server.getAddress().getPort())
+                            .path("/")
+                            .build();
+                    System.out.println("issuing request to " + reqURI);
+                    final HttpResponse<Void> resp;
+                    try {
+                        resp = client.send(HttpRequest.newBuilder(reqURI).build(),
+                                BodyHandlers.discarding());
+                        if (!presentClientCerts) {
+                            // request was expected to fail since the server was configured to force
+                            // the client to present the client cert, but the client didn't
+                            // present any
+                            fail("request was expected to fail, but didn't");
+                        }
+                        assertEquals(200, resp.statusCode(), "unexpected response code");
+                        // verify that the server asked the client to present the certificates
+                        assertTrue(clientSSLCtx.keyManager.clientAuthInitiated,
+                                "server was expected to request client certs, but didn't");
+                    } catch (IOException ioe) {
+                        if (presentClientCerts) {
+                            // wasn't expected to fail, just let the exception propagate
+                            throw ioe;
+                        }
+                        // verify that the server asked the client to present the certificates
+                        assertTrue(clientSSLCtx.keyManager.clientAuthInitiated,
+                                "server was expected to request client certs, but didn't");
+                        // verify it failed due to right reason
+                        Throwable cause = ioe;
+                        while (cause != null) {
+                            // either of SocketException or SSLHandshakeException are OK
```

Comment:
I have this on my TODO list and I plan to come back to this shortly after I am done with the CSR text. I was expecting this to be one specific type of exception but I'm seeing 2 different types depending on how I launch the test. Plus there's an open PR in security-libs area which is proposing some changes in the alert/exception raised from the TLS handshake layer when the client doesn't send certificates to the server. I need to review those changes to see what kind of impact if any that change will have here.